contracts/Committee.sol

```diff
@@ -0,0 +1,548 @@
+			newCommitteeSize++;
+		}
+		committeeSize = newCommitteeSize;
+		_refreshReadyForCommitteeCount();
+		_refreshOldestReadyToSyncStandbyPos();
+		return (prevCommitteeSize, newCommitteeSize);
+	}
+
+	function _refreshReadyForCommitteeCount() private returns (uint, uint) {
+		uint newCount = readyForCommitteeCount;
+		uint prevCount = newCount;
+		while (newCount > 0 && (topology.length < newCount || !isReadyForCommittee(topology[newCount - 1]))) {
+			newCount--;
+		}
+		while (topology.length > newCount && isReadyForCommittee(topology[newCount])) {
+			newCount++;
+		}
+		readyForCommitteeCount = newCount;
+		return (prevCount, newCount);
+	}
+
+	function _refreshOldestReadyToSyncStandbyPos() private {
+		uint256 oldestTimestamp = uint(-1);
+		uint256 oldestPos;
+		for (uint i = committeeSize; i < topology.length; i++) {
+			uint t = members[topology[i]].readyToSyncTimestamp;
+			if (t < oldestTimestamp) {
+				oldestTimestamp = t;
+				oldestPos = i;
+			}
+		}
+		oldestReadyToSyncStandbyPos = oldestPos;
+	}
+
+	function _compareValidatorsByData(address v1, uint256 v1Weight, bool v1Ready, address v2, uint256 v2Weight, bool v2Ready) private pure returns (int) {
+		return v1Ready && !v2Ready ||
+				v1Ready == v2Ready  && v1Weight > v2Weight ||
+				v1Ready == v2Ready  && v1Weight == v2Weight && uint256(v1) > uint256(v2)
+		? int(1) : -1;
+	}
+
+	function _compareValidators(address v1, address v2) private view returns (int) {
+		bool v1Ready = isReadyForCommittee(v1);
+		uint256 v1Weight = getValidatorWeight(v1);
+		bool v2Ready = isReadyForCommittee(v2);
+		uint256 v2Weight = getValidatorWeight(v2);
+		return _compareValidatorsByData(v1, v1Weight, v1Ready, v2, v2Weight, v2Ready);
+	}
+
+	function _replace(uint p1, uint p2) private {
+		address tempValidator = topology[p1];
+		topology[p1] = topology[p2];
+		topology[p2] = tempValidator;
+	}
+
+	function _repositionTopologyMember(uint memberPos) private returns (uint newPos, uint prevCommitteeSize, uint newCommitteeSize, uint prevStandbySize, uint newStandbySize) {
+		uint topologySize = topology.length;
+		assert(topologySize > memberPos);
+
+		while (memberPos > 0 && _compareValidators(topology[memberPos], topology[memberPos - 1]) > 0) {
+			_replace(memberPos-1, memberPos);
+			memberPos--;
+		}
+
+		while (memberPos < topologySize - 1 && _compareValidators(topology[memberPos + 1], topology[memberPos]) > 0) {
+			_replace(memberPos, memberPos+1);
+			memberPos++;
+		}
+
+		newPos = memberPos;
+
+		(prevCommitteeSize, newCommitteeSize) = _onTopologyModification();
+
+		prevStandbySize = topologySize - prevCommitteeSize;
```

Comment:
BUG - if toplogy increased `topologySize` is not longer valid for this computation, it needs the original topology size. This can result in prevStandbySize being larger than it should be.